src/datasources/exchange-api/exchange-api.service.spec.ts

```diff
@@ -0,0 +1,45 @@
+import { ExchangeApi } from './exchange-api.service';
+import { ExchangeResult } from '../../domain/exchange/entities/exchange-result.entity';
+import { FiatCodesExchangeResult } from '../../domain/exchange/entities/fiat-codes-result.entity';
+import { FakeConfigurationService } from '../../common/config/__tests__/fake.configuration.service';
+import { mockNetworkService } from '../../datasources/network/__tests__/test.network.module';
+
+const FIATCODES: FiatCodesExchangeResult = {
+  success: true,
+  symbols: {},
+};
+
+const EXCHANGERESULT: ExchangeResult = {
+  success: true,
+  rates: {},
+  base: '',
+};
+
+describe('ExchangeApi', () => {
+  const fakeConfigurationService = new FakeConfigurationService();
+  fakeConfigurationService.set('exchange.baseUri', 'http://exchange.test');
+  fakeConfigurationService.set('exchange.apiKey', 'aaaaaaaabbbbbbb');
+  const service: ExchangeApi = new ExchangeApi(
+    fakeConfigurationService,
+    mockNetworkService,
+  );
```

Comment:
Here the instance will be reused across tests in this file. This can cause issues if we have a state stored in `ExchangeApi`.

It is better to create and assign the instance in `beforeEach`.

You should also use `jest.clearAllMocks()` so that any mocked state is cleared.